pytorch_forecasting/models/tcn/__init__.py

```diff
@@ -0,0 +1,179 @@
+        x_categoricals: List[str],
+        embedding_sizes: Dict[str, Tuple[int, int]],
+        embedding_labels: Dict[str, List[str]],
+        static_categoricals: List[str],
+        static_reals: List[str],
+        time_varying_categoricals_encoder: List[str],
+        time_varying_categoricals_decoder: List[str],
+        time_varying_reals_encoder: List[str],
+        time_varying_reals_decoder: List[str],
+        embedding_paddings: List[str],
+        categorical_groups: Dict[str, List[str]],
+        
+        **kwargs,
+    ):
+        # saves arguments in signature to `.hparams` attribute, mandatory call - do not skip this
+        self.save_hyperparameters()
+        
+        # pass additional arguments to BaseModel.__init__, mandatory call - do not skip this
+        super().__init__(**kwargs)
+
+        # create embedder - can be fed with x["encoder_cat"] or x["decoder_cat"] and will return
+        # dictionary of category names mapped to embeddings
+        self.input_embeddings = MultiEmbedding(
+            embedding_sizes=self.hparams.embedding_sizes,
+            categorical_groups=self.hparams.categorical_groups,
+            embedding_paddings=self.hparams.embedding_paddings,
+            x_categoricals=self.hparams.x_categoricals,
+        )
+
+        # calculate the size of all concatenated embeddings + continous variables
+        n_features = sum(
+            embedding_size for classes_size, embedding_size in self.hparams.embedding_sizes.values()
+        ) + len(self.reals)
+
+        # create network that will be fed with continious variables and embeddings
+        self.network = nn.Sequential(TemporalConvNet(
+            num_inputs=n_features,
+            num_channels=self.hparams.n_hidden_layers,
+            kernel_size=self.hparams.kernel_size,
+            dropout=self.hparams.conv_dropout),
+            GAP1d(),
+            nn.Dropout(self.hparams.fc_dropout),
+            nn.Linear(self.hparams.n_hidden_layers[-1],self.hparams.output_size)
+            
+            
+        )
+
+
+    def forward(self, x: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
+        # x is a batch generated based on the TimeSeriesDataset
+        batch_size = x["encoder_lengths"].size(0)
+        embeddings = self.input_embeddings(x["encoder_cat"])  # returns dictionary with embedding tensors
+        network_input = torch.cat(
+            [x["encoder_cont"]]
+            + [
+                emb
+                for name, emb in embeddings.items()
+                if name in self.encoder_variables or name in self.static_variables
+            ],
+            dim=-1,
+        )
+        prediction = self.network(network_input.permute(0,2,1))
+        
+
+        # We need to return a dictionary that at least contains the prediction and the target_scale.
+        # The parameter can be directly forwarded from the input.
+        return dict(prediction=prediction, target_scale=x["target_scale"])
```

Comment:
Could you use the new way to return de-normalized predictions?
```suggestion
        prediction = self.transform_output(prediction, target_scale=x["target_scale"])
        return self.to_network_output(prediction=prediction)
```